Tests/OpenCombineTests/PublisherTests/MapErrorTests.swift

```diff
@@ -0,0 +1,251 @@
+        mapError.subscribe(tracking)
+        // Then
+        XCTAssertEqual(subscription.history, [.requested(.max(expectedSubscribeDemand))])
+    }
+
+    func testDemandSend() {
+        // Given
+        let expectedReceiveValueDemand = 4
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let mapError = publisher.mapError { OtherError($0) }
+        let tracking = TrackingSubscriberBase<Int, OtherError>(
+            receiveValue: { _ in .max(expectedReceiveValueDemand) }
+        )
+        // When
+        mapError.subscribe(tracking)
+        // Then
+        XCTAssertEqual(publisher.send(0), .max(expectedReceiveValueDemand))
+    }
+
+    func testCompletion() {
+        // Given
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let mapError = publisher.mapError { OtherError($0) }
+        let tracking = TrackingSubscriberBase<Int, OtherError>(
+            receiveSubscription: { $0.request(.unlimited) }
+        )
+        // When
+        mapError.subscribe(tracking)
+        publisher.send(completion: .finished)
+        // Then
+        XCTAssertEqual(subscription.history, [.requested(.unlimited)])
+        XCTAssertEqual(
+            tracking.history,
+            [.subscription(Subscriptions.empty), .completion(.finished)]
+        )
+    }
+
+    func testCancel() throws {
+        // Given
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let mapError = publisher.mapError { OtherError($0) }
+        var downstreamSubscription: Subscription?
+        let tracking = TrackingSubscriberBase<Int, OtherError>(
+            receiveSubscription: {
+            $0.request(.unlimited)
+            downstreamSubscription = $0
+            })
+        // When
+        mapError.subscribe(tracking)
+        try XCTUnwrap(downstreamSubscription).cancel()
+        // Then
+        XCTAssertEqual(subscription.history, [.requested(.unlimited), .cancelled])
+    }
+
+    func testCancelAlreadyCancelled() throws {
+        // Given
+        let subscription = CustomSubscription()
+        let publisher = CustomPublisher(subscription: subscription)
+        let mapError = publisher.mapError { OtherError($0) }
+        var downstreamSubscription: Subscription?
+        let tracking = TrackingSubscriberBase<Int, OtherError>(
+            receiveSubscription: {
+            $0.request(.unlimited)
+            downstreamSubscription = $0
+            })
+        // When
+        mapError.subscribe(tracking)
+        try XCTUnwrap(downstreamSubscription).cancel()
+        downstreamSubscription?.request(.unlimited)
+        try XCTUnwrap(downstreamSubscription).cancel()
+        // Then
+        XCTAssertEqual(subscription.history, [.requested(.unlimited),
+                                              .cancelled])
```

Comment:
I have this test failing in compatibility mode:

```
Test Case '-[OpenCombineTests.MapErrorTests testCancelAlreadyCancelled]' started.
MapErrorTests.swift:179: error: -[OpenCombineTests.MapErrorTests testCancelAlreadyCancelled] : XCTAssertEqual failed: ("[requested(unlimited), cancelled, requested(unlimited), cancelled]") is not equal to ("[requested(unlimited), cancelled]")
Test Case '-[OpenCombineTests.MapErrorTests testCancelAlreadyCancelled]' failed (0.497 seconds).
```

We need to change this assert so it's not hit when testing against Combine, and tweak the implementation of `MapError` like here.

https://github.com/broadwaylamb/OpenCombine/blob/bb419867f1baaaf77a32103df5ab9975bfff4a9e/Sources/OpenCombine/Publishers/Publishers.Map.swift#L155-L158

It'd be very good if you could also add a test ensuring that if the upstream publisher sends failure _twice_, the downstream will also receive it twice. Or once, depending on how Apple Combine behaves. After that decide whether we need this `_transform = nil` line like in `Map`.

Reply:
I think this is working for me in compatibility mode.  Is it still failing for you? @broadwaylamb

Reply:
No, not anymore.

Reply:
@broadwaylamb I've added a another sent failure to extend the error.  Ready for review again.